source/UIABrowseMode.py

```diff
@@ -152,27 +152,34 @@ def isChild(self,parent):
 		return self.level>parent.level
 
 def UIAHeadingQuicknavIterator(itemType,document,position,direction="next"):
-	if position:
-		curPosition=position
+	reverse = bool(direction == "previous")
+	entireDocument = document.makeTextInfo(textInfos.POSITION_ALL)
+	if not position:

```

Comment:
When would this happen? Also, would a check for None be more appropriate here?

```suggestion
	if position is None:
```